content/departments/cloud/technical-docs/v2.0/disaster_recovery_process.md

```diff
@@ -0,0 +1,130 @@
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+- connect to cluster
+
+```sh
+mi2 instance workon -e $ENVIRONMENT --slug $SLUG
+# paste output into terminal
+```
+
+- verify node zone
+
+```sh
+kubectl get nodes
+kubectl describe node <NODE_FROM_CLUSTER> | grep zone
+```
+
+- perform zone failover (remove node zone from GKE node locations)
+
+```sh
+gcloud container node-pools list --cluster $GKE_NAME --region $GKE_REGION
+gcloud container node-pools describe primary --cluster $GKE_NAME --region $GKE_REGION --format json | jq '.locations'
+gcloud container node-pools update primary --cluster $GKE_NAME --region $GKE_REGION --node-locations <DIFFERENT_ZONE_THAN_EXISTING_NODE> --async
+```
+
+- verify pods were terminated
+
+```
+kubectl get pods # should show failing pods, b/c node was drained
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health # should fail
+```
+
+- wait for new node to be ready
+
+```sh
+kubect get nodes # waiting for new node
+```
+
+- verify new node zone
+
+```sh
+kubectl describe node <NEW_NODE> | grep zone # should be different from previous node
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+2. CloudSQL zone failover
+
+- export environment variables
+
+```sh
+export ENVIRONMENT=[dev|prod]
+export SLUG=<SLUG>
+export CLOUDSQL_INSTANCE_NAME=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.status.cloudSQL[0].name')
+export GCP_PROJECT=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.status.gcpProjectId')
+export INSTANCE_ID=$(mi2 instance get -e $ENVIRONMENT --slug $SLUG | jq -r '.metadata.name')
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+- patch CloudSQL instance to use different zone
+
+```sh
+gcloud sql instances describe $CLOUDSQL_INSTANCE_NAME --project $GCP_PROJECT | grep zone
+# returns actual CloudSQL zone <OLD_ZONE>
+gsed -i 's/  gcpZone: <OLD_ZONE>/  gcpZone: <FAILOVER_ZONE/g' environments/$ENVIRONMENT/deployments/$INSTANCE_ID/config.yaml
+mi2 generate cdktf -e $ENVIRONMENT --slug $SLUG
+cd environments/$ENVIRONMENT/deployments/$INSTANCE_ID/terraform/stacks/sql
+terraform init && terraform apply
+
+gcloud sql instances describe $CLOUDSQL_INSTANCE_NAME --project $GCP_PROJECT | grep zone
+# should return <FAILOVER_ZONE>
+cd -
+```
+
+- check instance is healthy
+
+```sh
+mi2 instance check --slug $SLUG -e $ENVIRONMENT pods-health
+curl -sSL --fail https://$SLUG.sourcegraph.com/sign-in -i
+```
+
+- restore backup in different zone
+
+```sh
+mi2 instance sql-backup list --slug $SLUG -e $ENVIRONMENT
+mi2 instance sql-restore create --backup-id $SQL_BACKUP_ID --slug $SLUG -e $ENVIRONMENT
+# wait until ready
+gcloud sql instances describe $CLOUDSQL_INSTANCE_NAME --project $GCP_PROJECT
+```
```

Comment:
is backup restoration always necessary as part of the zone failover process? I suppose we aren't expecting any data loss in the first place?

Reply:
It's not necessary, but in the worst case scenario we should assume that CloudSQL instance disk was lost - so I added optional scenario, which is testing that we can restore from backup in different zone (it proved that backup is regional and works). I will add OPTIONAL note.